### PR TITLE
Address todo comments

### LINKED ISSUE #355: Review TODOs / FIXMEs
Before the mainnet release version, let's review all the TODOs and FIXMEs in the code base (tgrade, poe-contracts, and tgrade-contracts, at least).

The outcome of this issue must be a referenced list of TODOs / FIXMEs, with some evaluation of their importance / requirement for Mainnet launch, and follow-up issues for the important cases.

A quick search:

todo/TODO:

- app/genesis_integration_test.go:	// todo (Alex): enable require.NoError(t, twasmGs.ValidateBasic())
- app/export.go:		WithBlockTime(time.Now().UTC()) // todo (Alex): check if there is any way to get the last block time
- scripts/protocgen.sh:--doc_opt=./docs/proto/protodoc-markdown.tmpl,proto-docs.md \
- x/poe/stakingadapter/keeper.go:	// todo: return self delegation of validator
- x/poe/stakingadapter/keeper.go:	// todo: handle for validator operator's self delegation
- x/poe/stakingadapter/keeper.go:	// todo: handle for validator operator's self delegation
- x/poe/stakingadapter/keeper.go:	// todo: handle for validators self delegations
- x/poe/module.go:	// todo: add PoE validation
- x/twasm/contract/incoming_msgs.go:		// todo (Alex): check how a client update is done now. this also needs to go into the contracts
- x/twasm/contract/incoming_msgs.go:				// todo (Alex): SubstituteClientId: proxy.ClientId,
- x/twasm/contract/incoming_msgs.go:			var proxy = struct { // todo: better use wasmvmtypes.MigrateMsg when names match
- x/twasm/contract/incoming_msgs_test.go:				// todo (Alex):fix also in contracts!!				SubstituteClientId:      ib,
- x/twasm/abci.go:		panic(err) // todo (reviewer): this will break consensus
- go.mod:	// TODO Remove it: https://github.com/cosmos/cosmos-sdk/issues/10409
- testing/fee_test.go:	// TODO: no more fees are distributed. Rather they are held in a new contract to be withdrawn.
- third_party/proto/confio/proofs.proto:    bytes key = 1; // TODO: remove this as unnecessary??? we prove a range
- third_party/proto/confio/proofs.proto:  bytes key = 1; // TODO: remove this as unnecessary??? we prove a range
- x/poe/contract/tg4_stake.go:// TODO: add pagination support here!
- x/poe/contract/tgrade_valset.go:	// TODO: add this if you want it, not in current code
- x/poe/contract/tgrade_valset.go:// TODO: add pagination support
- x/poe/contract/tgrade_valset.go:	// TODO: this is just a placeholder trying to get 100
- x/poe/contract/tgrade_oc_proposals.go:	// TODO: clarify this format
- x/poe/bootstrap.go:		// TODO: allow to configure the other types.

### PR DESCRIPTION
**Pull request by alpe**
Resolves #355 

Fixed or moved todo into new github issues

Note integration test todo is fixed in #361

### CODE REVIEW

File: testing/fee_test.go [hunk 1]
```diff
@@ -46,25 +50,35 @@ func TestFeeDistribution(t *testing.T) {
...
-	// TODO: no more fees are distributed. Rather they are held in a new contract to be withdrawn.
-	// DO proper fix in issue #156, so we can query the pending stake. For now I will disable
-	// then
-	//for i := 0; i < sut.nodesCount; i++ {
-	//	newBalance := cli.QueryBalance(cli.GetKeyAddr(fmt.Sprintf("node%d", i)), "utgd")
-	//	diff := newBalance - oldBalances[i]
-	//	t.Logf("Block rewards: %d\n", diff)
-	//	assert.LessOrEqualf(t, int64(20000000/sut.nodesCount), diff, "got diff: %d (before %d after %d)", diff, oldBalances[i], newBalance)
-	//}
+	// and rewards claimed
```
Comment (maurolacy) [inline testing/fee_test.go:65]: 👍🏼 Fantastic.

File: x/twasm/abci.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func BeginBlocker(parentCtx sdk.Context, k abciKeeper, b abci.RequestBeginBlock)
 
 	msgBz, err := json.Marshal(msg)
 	if err != nil {
-		panic(err) // todo (reviewer): this will break consensus
```
Comment (maurolacy) [inline x/twasm/abci.go:60]: Wouldn't make sense to leave this, just as a warning?

**Review by maurolacy — APPROVED** [review]
Looks good. Put some some comments, but it's your call.

File: go.mod [hunk 3]
```diff
@@ -128,7 +128,7 @@ replace (
 	// Use the cosmos-flavored keyring library
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
 	// Fix upstream GHSA-h395-qcrw-5vmq vulnerability.
-	// TODO Remove it: https://github.com/cosmos/cosmos-sdk/issues/10409
+	// Remove it: https://github.com/cosmos/cosmos-sdk/issues/10409
```
[the remark this answers is not in the document]
Comment (alpe) [reply go.mod:131]: This todo is not actionable currently.

File: x/twasm/abci.go (hunk 2 above)
Comment (alpe) [reply x/twasm/abci.go:60]: updated the text

**Pull request merged**